### PR TITLE
Improvements for automation

### PR DESCRIPTION
**Pull request by valentin-lup**
Added a new cfg file for the menu functions (menu_functions.cfg)
Added the upgrade_squad functions (fix brought from the mainnet scripts)
Allowed the use of the command line arguments besides the menu options

### CODE REVIEW

File: config/menu_functions.cfg [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+       fi
+}
+
+function upgrade_squad {
+  paths
+  #Check to see if scripts have been updated
+  cd $SCRIPTPATH
+  CURRENT_SCRIPTS_COMMIT=$(git show | grep -m 1 commit | awk '{print $2}')
+  
+      if [ $LATEST_SCRIPTS_COMMIT == $CURRENT_SCRIPTS_COMMIT ]; then
```
Comment (iulianpascalau) [inline config/menu_functions.cfg:159]: same here, we need to allow continuing even if the scripts are outdated

File: script.sh [hunk 2]
```diff
@@ -12,446 +12,169 @@ SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
...
-  echo -e
-  break
+  get_logs
   ;;
-
 esac
 
-done
\ No newline at end of file
+fi
```
Comment (iulianpascalau) [inline script.sh:180]: add empty line

File: config/menu_functions.cfg [hunk 3]
```diff
@@ -0,0 +1,438 @@
+#!/bin/bash
+set -e
+
+source $SCRIPTPATH/config/variables.cfg
+source $SCRIPTPATH/config/functions.cfg
+
+function menu_install {
```
Comment (iulianpascalau) [inline config/menu_functions.cfg:7]: maybe remove all "menu" strings from all function names?

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: config/menu_functions.cfg (hunk 3 above)
Comment (valentin-lup) [reply config/menu_functions.cfg:7]: Can't. There is another function named "install"... This way I know in which file this function resides...

File: script.sh (hunk 2 above)
Comment (valentin-lup) [reply script.sh:180]: Done !

**Review by andreibancioiu — CHANGES_REQUESTED** [review]
Only what Iulian has said.

Optionally, do you think we can remove the `github_pull` and github checks all together, and just instruct all users to do `git pull` before each operation (just an opinion)?

**Comment by iulianpascalau** [comment]
> Only what Iulian has said.
> 
> Optionally, do you think we can remove the `github_pull` and github checks all together, and just instruct all users to do `git pull` before each operation (just an opinion)?

We have agreed that the latest version checking should be done as the very first step when we call `script.sh`

File: config/menu_functions.cfg [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+                  echo -e "${GREEN}Fine ! Skipping upgrade on this machine...${NC}"
+                  ;;
+            
+                * )
+                  echo -e "${GREEN}I'll take that as a no then... moving on...${NC}"
+                  ;;
+            esac
+  
+        else
+          echo -e "${RED}You do not have the latest version of the elrond-go-scripts-testnet !!!${NC}"
```
[the remark this answers is not in the document]
Comment (valentin-lup) [reply config/menu_functions.cfg]: Fixed.

File: config/menu_functions.cfg (hunk 3 above)
Comment (valentin-lup) [reply config/menu_functions.cfg]: Fixed this as intended by the reviewer ;) (normalised function names)

File: config/menu_functions.cfg (hunk 1 above)
Comment (valentin-lup) [reply config/menu_functions.cfg]: Solved by adding just a global check bypass-able by the user.

**Review by iulianpascalau — COMMENTED** [review]
Testing in progress...

**Review by andreibancioiu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**